packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1673,6 +1676,60 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.context.getAbsoluteUrl(relativeUrl);
     }
 
+    /**
+     * Find and reset nodes whose unreferenced state, such as unreferenced time, needs to be reset. These are nodes
+     * that transitioned from `unreferenced -> referenced -> unreferenced` since the last GC run.
+     * @param resultFromCurrentRun - The result of the current GC run.
+     */
+    private resetUnreferencedState(resultFromCurrentRun: IGCResult) {
+        assert(this.gcDataFromLastRun !== undefined, "GC data from last run should be available");
+
+        // Get the list of references that were added since the last GC run.
+        // TODO: The cast to any is done temporarily until the change to IFluidHandleContext in `core-interfaces` is
+        // merged and client is updated to the new `core-interfaces` version.
+        const outboundRoutesSinceLastGC: string[] = (this.IFluidHandleContext as any).decodedHandles;
+
+        /**
+         * Run GC on the data from the last run to get the reference graph starting from the above routes.
+         * Note that this has to be done on the reference graph from the last run because routes added since then may
+         * have had outbound references that have been removed now. We need to get a list of all those references.
+         * For example, consider the following scenario where A, B and C are data stores:
+         * 1. Last summary happened at t1. Reference graph: A    B (t1) -> C (t1). B and C have unreferenced time t1.
```

Comment:
It was not clear for me what you mean here initially.
The reason - you are listing nodes and edges in same list, which is a bit confusing (even comma after A would help).
I'd rather see something more closer to how it's represented in math (see [example](https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)#Examples)):
V = [A, B, C], E = [{B, C}] (they are sets, so in math we use {}, not [] :))
For readability we can describe it as E = [B -> C], but E has no A as there is no edges from A.